### PR TITLE
Add tests for upsell banner feature

### LINKED ISSUE #3797: Write test(s) for #3748
Write test for PR #3748 

Add a unit test that verifies that the banner is shown (and others are hidden) in the right circumstances.

### PR DESCRIPTION
**Pull request by lloan**
This PR fixes #3797 

# How to test
- [x] Are we showing free users a similar welcome and metrics like premium users see that include (email masks used, emails blocked, emails forwarded, and trackers removed) `This is already tested on the file by previous committer`
- [X] When a user hits their 5 mask limit, are we showing the upsell header 
- [X] Upgrade to premium CTA takes user to premium price plans
- [X] Are we providing US users the right experience (correct banners, links, etc.)
- [X] Are we providing non-US users the right experience (correct banners, links, etc.)
- [X] Are we showing a tooltip when the user hovers over the number of masks they have after they reach the limit?

# Checklist (Definition of Done)
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] I've added or updated relevant docs in the docs/ directory
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.
- [ ] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] l10n changes have been submitted to the l10n repository, if any.

### CODE REVIEW

**Comment by lloan** [comment]
PR opened to run CCI test suite. @raffifu great job! I tested this and it looks good to me.

File: frontend/src/pages/accounts/profile.test.tsx [hunk 1]
```diff
@@ -449,6 +450,151 @@ describe("The dashboard", () => {
...
+    setMockAliasesDataOnce({ random: [{ enabled: true, id: 42 }], custom: [] });
+    setMockProfileDataOnce({ has_premium: false });
+    setMockRuntimeDataOnce(getMockRuntimeDataWithPeriodicalPremium());
+    const mockedConfig = mockConfigModule.getRuntimeConfig();
+    // getRuntimeConfig() is called frequently, so mock its return value,
+    // then restore the original mock at the end of this test:
+    mockConfigModule.getRuntimeConfig.mockReturnValue({
+      ...mockedConfig,
+      maxFreeAliases: 1,
+    });
```
Comment (groovecoder) [inline frontend/src/pages/accounts/profile.test.tsx:582]: suggestion (non-blocking): This is the 4th copy-paste of this test setup code. Refactor it into a separate function? Either here in this test, or maybe "up" in a module in `frontend/__mocks__/hooks/api` somewhere?

(Note: [the other test setup](https://github.com/mozilla/fx-private-relay/pull/4038/files#diff-eb5b70afbfb6260a42fd6afd04933e7c4d95ef09d2dd67907dd11e08e0947360R478) calls `getMockRuntimeDataWithPhones` so it still needs its own custom setup code.

**Review by groovecoder — APPROVED** [review]

File: frontend/src/pages/accounts/profile.test.tsx (hunk 1 above)
Comment (lloan) [reply frontend/src/pages/accounts/profile.test.tsx]: done

File: frontend/src/pages/accounts/profile.test.tsx [hunk 2]
```diff
@@ -449,6 +450,151 @@ describe("The dashboard", () => {
...
+
+    const upsellBanner = screen.queryByRole("link", {
+      name: "l10n string: [profile-maxed-aliases-cta], with vars: {}",
+    });
+
+    expect(upsellBanner).toBeInTheDocument();
+    mockConfigModule.getRuntimeConfig.mockReturnValue(mockedConfig);
+  });
+
+  it("shows correct upsell banner for US country", () => {
```
[the remark this answers is not in the document]
Comment (lloan) [reply frontend/src/pages/accounts/profile.test.tsx]: done

**Pull request merged**